### PR TITLE
Fix wiping sensitive info in logs

### PR DESCRIPTION
**Pull request by vitlibar**
### Changelog category:
- Bug Fix

### Changelog entry:
Fix wiping sensitive info in logs

### CODE REVIEW

File: src/Parsers/ParserInsertQuery.cpp [hunk 1]
```diff
@@ -256,14 +256,23 @@ bool ParserInsertQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
...
+            query->children.push_back(compression);
     }
 
     if (table_function)
     {
         query->table_function = table_function;
         query->partition_by = partition_by_expr;
+
+        if (table_function)
+            query->children.push_back(table_function);
```
Comment (vitlibar) [inline src/Parsers/ParserInsertQuery.cpp:273]: Here was a problem. `ParserInsertQuery` didn't add `table_function` as its child and thus [IAST::childrenHaveSecretParts()](https://github.com/ClickHouse/ClickHouse/blob/0dcb98565b471f2c5fc715c3498d5a4d5a311e04/src/Parsers/IAST.cpp#L185) didn't understand when an INSERT INTO TABLE statement has secret parts.

File: src/Parsers/ParserInsertQuery.cpp [hunk 2]
```diff
@@ -256,14 +256,23 @@ bool ParserInsertQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
     if (infile)
     {
         query->infile = infile;
+        query->compression = compression;
+
+        if (infile)
```
Comment (SmitaRKulkarni) [inline src/Parsers/ParserInsertQuery.cpp:261]: this `if` can be removed

File: src/Parsers/ParserInsertQuery.cpp [hunk 3]
```diff
@@ -256,14 +256,23 @@ bool ParserInsertQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
...
-            query->compression = compression;
+            query->children.push_back(compression);
     }
 
     if (table_function)
     {
         query->table_function = table_function;
         query->partition_by = partition_by_expr;
+
+        if (table_function)
```
Comment (SmitaRKulkarni) [inline src/Parsers/ParserInsertQuery.cpp:272]: same as above, can be removed

**Review by SmitaRKulkarni — COMMENTED** [review]
Rest all LGTM

File: src/Parsers/ParserInsertQuery.cpp (hunk 2 above)
Comment (vitlibar) [reply src/Parsers/ParserInsertQuery.cpp]: done

**Review by SmitaRKulkarni — APPROVED** [review]

**Pull request merged**